### DISCUSSION TITLE
postgres user using postgres database

### DISCUSSION THREAD

**Discussion by D3an1el** [madeofpendletonwool/PinePods#353] [Q&A]
Hi

is it possible to setup without adding the user to "postgres" database? The user has all priviliges to the pinepod database?

on a different note can a uri be added instead of using environment variables so i can use certs to connect to my postgres instances ?

```
pinepods  | connection to server at "192.168.1.180", port 5432 failed: FATAL:  no pg_hba.conf entry for host "192.168.1.149", user "pinepods", database "postgres", no encryption
```

**Comment by madeofpendletonwool** [comment]
Hi @D3an1el,
To your first question, yes, it should be. The issue is that with the default postgres container we pull down the only user that has permission to create dbs and setup tables is the postgres user. So in order to make this work one of two things would need to happen:

1. The user would need to setup a user with fine grained permission for a particular db user. From there, it should work fine. So if I set
services:
  post-db:
    image: postgres:latest
    environment:
      POSTGRES_DB: pinepods_database
      POSTGRES_USER: pinepods

that *should* work given that I have already gone and setup the user permission for the pinepods user to access the already created pinepods_database. Of course for Pinepods vars I would need to then set my DB_USER var to that as well

DB_USER: pinepods

Or 2. You would need to pass both your user without additional privileges AND a privileged db user. One to create the db and setup a user and permissions, and one the app actually runs off. Which, at that point it defeats the purpose because you've already passed your privileged user in a var to the pinepods container anyway.  

So based on that, I think option 1 with some decent documentation for users who'd like to take the extra step of creating a user with locked down access would be the way to go. Most people likely wouldn't want to do it, but, I'm all for supporting additional security settings for those who'd like to spend the time implementing them. 

As for your second question on uri's. Not currently, though I do think it's worth opening up an issue to change the database value parsing to handle everything as uri's. Even if you pass the individual values the server would build a proper uri out of them. That would allow either uri's or individual values.

**Comment by D3an1el** [comment]
Thanks for your response. With my instances i have made "pinepods" owner of the db "pinepods_database", so i would assume with option 1 it should of worked?.

 Inregards to URI it would allow for secure [TCP/IP connection](https://www.postgresql.org/docs/8.4/libpq-connect.html#LIBPQ-CONNECT-SSLMODE), but would need to allow for self signed certs for my use case as i use selfsigned with postgres connections when possible.

**Comment by D3an1el** [comment]
So i assume i will have to wait until URI is implemented so it allows me to use only one privileged user ?
